lib/src/editor.dart

```diff
@@ -0,0 +1,790 @@
+
+    if (!deepEquals(actualNode, expectedNode)) {
+      throw Exception('''
+Modification did not result in expected result! 
+Obtained: 
+$actualNode
+Expected: 
+$expectedNode''');
+    }
+
+    edits.add(edit);
+  }
+}
+
+/// Performs the string operation on [yaml] to achieve the effect of setting
+/// the element at [index] to [newValue] when re-parsed.
+SourceEdit _setInList(
+    String yaml, YamlList list, int index, Object newValue, YamlStyle style) {
+  final currValue = list.nodes[index];
+
+  final offset = currValue.span.start.offset;
+  final length = currValue.span.end.offset - offset;
+
+  return SourceEdit(offset, length, newValue.toString());
+}
+
+/// Performs the string operation on [yaml] to achieve the effect of removing
+/// the element at [index] when re-parsed.
+///
+/// Returns the node that is removed.
+SourceEdit _removeInList(String yaml, YamlList list, int index) {
+  final nodeToRemove = list.nodes[index];
+
+  if (list.style == CollectionStyle.FLOW) {
+    return _removeFromFlowList(yaml, list, nodeToRemove, index);
+  } else {
+    return _removeFromBlockList(yaml, list, nodeToRemove, index);
+  }
+}
+
+/// Performs the string operation on [yaml] to achieve the effect of
+/// removing [elem] when re-parsed.
+///
+/// Returns `true` if [elem] was successfully found and removed, `false` otherwise.
+SourceEdit _removeFromList(String yaml, YamlList list, Object elem) {
+  var index = _indexOf(list, elem);
+  if (index == -1) return null;
+
+  return _removeInList(yaml, list, index);
+}
+
+/// Performs the string operation on [yaml] to achieve the effect of
+/// appending [elem] to the list.
+SourceEdit _addToList(
+    String yaml, YamlList list, Object elem, YamlStyle style) {
+  if (list.style == CollectionStyle.FLOW) {
+    return _addToFlowList(yaml, list, elem, style);
+  } else {
+    return _addToBlockList(yaml, list, elem, style);
+  }
+}
+
+/// Performs the string operation on [yaml] to achieve the effect of
+/// prepending [elem] to the list.
+SourceEdit _prependToList(
+    String yaml, YamlList list, Object elem, YamlStyle style) {
+  if (list.style == CollectionStyle.FLOW) {
+    return _prependToFlowList(yaml, list, elem, style);
+  } else {
+    return _prependToBlockList(yaml, list, elem, style);
+  }
+}
+
+/// Performs the string operation on [yaml] to achieve a similar effect of
+/// inserting [elem] to the list at [index].
+SourceEdit _insertInList(
+    String yaml, YamlList list, int index, Object elem, YamlStyle style) {
+  if (index > list.length || index < 0) {
+    throw RangeError.range(index, 0, list.length);
+  }
+
+  /// We call the add method if the user wants to add it to the end of the list
+  /// because appending requires different techniques.
+  if (index == list.length) {
+    return _addToList(yaml, list, elem, style);
+  } else {
+    if (list.style == CollectionStyle.FLOW) {
+      return _insertInFlowList(yaml, list, index, elem, style);
```

Comment:
There are a lot of different code-paths to cover different yaml-constructs. It might be worth considering setting up coverage for the testing to make sure all/most code-paths are tested.

Reply:
Definitely! Will implement soon!